core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/udf/python/PythonUDFOpDescV2.scala

```diff
@@ -48,53 +51,98 @@ class PythonUDFOpDescV2 extends OperatorDescriptor {
   @JsonSchemaTitle("Python script")
   @JsonPropertyDescription("Input your code here")
   var code: String = ""
+
   @JsonProperty(required = true)
   @JsonSchemaTitle("Worker count")
   @JsonPropertyDescription("Specify how many parallel workers to lunch")
   var workers: Int = Int.box(1)
+
   @JsonProperty(required = true, defaultValue = "true")
   @JsonSchemaTitle("Retain input columns")
   @JsonPropertyDescription("Keep the original input columns?")
   var retainInputColumns: Boolean = Boolean.box(false)
+
   @JsonProperty
   @JsonSchemaTitle("Extra output column(s)")
   @JsonPropertyDescription(
     "Name of the newly added output columns that the UDF will produce, if any"
   )
   var outputColumns: List[Attribute] = List()
 
-  override def operatorExecutor(operatorSchemaInfo: OperatorSchemaInfo) = {
+  override def operatorExecutor(operatorSchemaInfo: OperatorSchemaInfo): OpExecConfig = {
     Preconditions.checkArgument(workers >= 1, "Need at least 1 worker.", Array())
+    val opInfo = this.operatorInfo
+    val partitionRequirement: List[Option[PartitionInfo]] = if (inputPorts != null) {
+      inputPorts.map(p => Option(p.partitionRequirement))
+    } else {
+      opInfo.inputPorts.map(_ => None)
+    }
+    val dependency: Map[Int, Int] = if (inputPorts != null) {
+      inputPorts
+        .flatMap(p => p.dependencies.map(dependee => p.portID.split("-")(1).toInt -> dependee))
```

Comment:
using portID to check dependency is not reliable, just use the port index in the list

```
      inputPorts.zipWithIndex.flatMap{
        case (port, i) => port.dependencies.map(dependee => i -> dependee)
      }.toMap
```

Reply:
done